pkg/collector/runner/runner.go

```diff
@@ -285,3 +290,30 @@ func (r *Runner) StopCheck(id checkid.ID) error {
 		return fmt.Errorf("timeout during stop operation on check id %s", id)
 	}
 }
+
+func (r *Runner) logWorkerUtilization() {
+	overview, err := r.utilizationMonitor.GetWorkerOverview()
+	if err != nil {
+		log.Warnf("Error getting worker utilization data: %v", err)
+		return
+	}
+
+	averageUtilization := fmt.Sprintf("%.3f", overview.AverageUtilization)
+	log.Debugf("Average worker utilization: %v", averageUtilization)
+
+	if len(overview.WorkersOverThreshold) > 0 {
+		log.Warnf("Workers over utilization threshold: %v", overview.WorkersOverThreshold)
```

Comment:
If this runs every minute we might need a log limiter to avoid printing the warning every single time